app/containers/EmojiPicker/index.tsx

```diff
@@ -1,106 +1,92 @@
-		};
-	}
-
-	async componentDidMount() {
-		await this.updateFrequentlyUsed();
-		this.setState({ show: true });
-	}
-
-	shouldComponentUpdate(nextProps: IEmojiPickerProps, nextState: IEmojiPickerState) {
-		const { frequentlyUsed, show, width } = this.state;
-		const { theme } = this.props;
-		if (nextProps.theme !== theme) {
-			return true;
-		}
-		if (nextState.show !== show) {
-			return true;
-		}
-		if (nextState.width !== width) {
-			return true;
-		}
-		if (!dequal(nextState.frequentlyUsed, frequentlyUsed)) {
-			return true;
-		}
-		return false;
-	}
+import { useTheme } from '../../theme';
+import { IEmoji, ICustomEmojis, TFrequentlyUsedEmojiModel } from '../../definitions';
+import { useAppSelector } from '../../lib/hooks';
+import { IEmojiPickerProps, EventTypes } from './interfaces';
+
+export const useFrequentlyUsedEmoji = () => {
+	const [frequentlyUsed, setFrequentlyUsed] = useState<(string | IEmoji)[]>([]);
+	const [loaded, setLoaded] = useState(false);
+	const getFrequentlyUsedEmojis = async () => {
+		const db = database.active;
+		const frequentlyUsedRecords = await db.get('frequently_used_emojis').query().fetch();
+		const frequentlyUsedOrdered = orderBy(frequentlyUsedRecords, ['count'], ['desc']);
+		const frequentlyUsedEmojis = frequentlyUsedOrdered.map(item => {
+			if (item.isCustom) {
+				return { content: item.content, extension: item.extension, isCustom: item.isCustom };
+			}
+			return shortnameToUnicode(`${item.content}`);
+		}) as (string | IEmoji)[];
+		setFrequentlyUsed(frequentlyUsedEmojis);
+		setLoaded(true);
+	};
+	useEffect(() => {
+		getFrequentlyUsedEmojis();
+	}, []);
+	return { frequentlyUsed, loaded };
+};
+
+const EmojiPicker = ({
+	baseUrl,
+	onItemClicked,
+	tabEmojiStyle,
+	isEmojiKeyboard = false,
+	searching = false,
+	searchedEmojis = []
+}: IEmojiPickerProps): React.ReactElement | null => {
+	const [width, setWidth] = useState(null);
+	const { colors } = useTheme();
+	const { frequentlyUsed, loaded } = useFrequentlyUsedEmoji();
+
+	const allCustomEmojis: ICustomEmojis = useAppSelector(state => state.customEmojis);
+	const customEmojis = useMemo(
+		() =>
+			Object.keys(allCustomEmojis)
+				.filter(item => item === allCustomEmojis[item].name)
+				.map(item => ({
+					content: allCustomEmojis[item].name,
+					extension: allCustomEmojis[item].extension,
+					isCustom: true
+				})),
+		[allCustomEmojis]
+	);
 
-	onEmojiSelected = (emoji: IEmoji) => {
+	const handleEmojiSelect = (emoji: IEmoji) => {
 		try {
-			const { onEmojiSelected } = this.props;
 			if (emoji.isCustom) {
-				this._addFrequentlyUsed({
+				_addFrequentlyUsed({
 					content: emoji.content,
 					extension: emoji.extension,
 					isCustom: true
 				});
-				onEmojiSelected(`:${emoji.content}:`);
+				onItemClicked(EventTypes.EMOJI_PRESSED, `:${emoji.content}:`);
 			} else {
 				const content = emoji;
-				this._addFrequentlyUsed({ content, isCustom: false });
+				_addFrequentlyUsed({ content, isCustom: false });
```

Comment:
```suggestion
				_addFrequentlyUsed({ content: emoji, isCustom: false });
```
remove the upper line